log-ballerina/natives.bal

```diff
@@ -16,135 +16,76 @@
 #
-# + msg - The message to be logged
-public isolated function printDebug(anydata|(isolated function () returns (anydata)) msg) {
-    if (isLogLevelEnabled(DEBUG)) {
-        if (msg is isolated function () returns (anydata)) {
-            logMessage(DEBUG, msg());
-        } else {
-            logMessage(DEBUG, msg);
-        }
-    }
-}
+# + msg - msg which cannot be a key
+public type KeyValues record {|
+    never msg?;
+    Value...;
+|};
 
-# Logs the specified message at ERROR level.
-# ```ballerina
-# error e = error("error occurred");
-# log:printError("error log with cause", err = e);
-# ```
-# 
-# + msg - The message to be logged
-# + err - The error struct to be logged
-public isolated function printError(anydata|(isolated function () returns (anydata)) msg, error? err = ()) {
-    if (isLogLevelEnabled(ERROR)) {
-        if (msg is isolated function () returns (anydata)) {
-            logMessageWithError(ERROR, msg(), err);
-        } else {
-            logMessageWithError(ERROR, msg, err);
-        }
-    }
-}
+# Key-Value pairs that needs to be desplayed in the error log.
+#
+# + msg - msg which cannot be a key
+# + err - error
+public type ErrorKeyValues record {|
+    never msg?;
+    never err?;
+    Value...;
+|};
 
-# Logs the specified message at INFO level.
+# Prints logs.
 # ```ballerina
-# log:printInfo("info log");
+# log:print("something went wrong", id = 845315)
 # ```
-# 
+#
 # + msg - The message to be logged
-public isolated function printInfo(anydata|(isolated function () returns (anydata)) msg) {
-    if (isLogLevelEnabled(INFO)) {
-        if (msg is isolated function () returns (anydata)) {
-            logMessage(INFO, msg());
+# + keyValues - The key-value pairs to be logged
+public isolated function print(string msg, *KeyValues keyValues) {
+    string keyValuesString = "";
+    foreach [string, Value] [k, v] in keyValues.entries() {
+        if (v is string) {
+            keyValuesString = keyValuesString + ", " + k + " = " + "\"" + v.toString() + "\"";
         } else {
-            logMessage(INFO, msg);
+            keyValuesString = keyValuesString + ", " + k + " = " + v.toString();
         }
     }
+    printExtern("message = " + "\"" + msg + "\"" + keyValuesString);
 }
 
-# Logs the specified message at TRACE level.
+# Prints error logs.
 # ```ballerina
-# log:printTrace("trace log");
+# error e = error("error occurred");
+# log:printError("error log with cause", err = e, id = 845315);
 # ```
 # 
 # + msg - The message to be logged
-public isolated function printTrace(anydata|(isolated function () returns (anydata)) msg) {
-    if (isLogLevelEnabled(TRACE)) {
-        if (msg is isolated function () returns (anydata)) {
-            logMessage(TRACE, msg());
+# + keyValues - The key-value pairs to be logged
+# + err - The error struct to be logged
+public isolated function printError(string msg, *ErrorKeyValues keyValues, error? err = ()) {
```

Comment:
```suggestion
public isolated function printError(string msg, error? err = (), *ErrorKeyValues keyValues) {
```

Reply:
required parameter not allowed after defaultable parameters